fetch.bs

```diff
@@ -5630,6 +5624,71 @@ if the script checks that the URL has the right hostname.
 
 
 
+<h2 id=data-urls><code>data:</code> URLs</h2>
+
+<p>For an informative description of <code>data:</code> URLs, see RFC 2397. This section replaces
+that RFC's normative processing requirements. [[RFC2397]]
+
+<p>A <dfn><code>data:</code> URL struct</dfn> is a <a>struct</a> that consists of a
+<dfn for="data: URL struct">MIME type</dfn> (a <a for=/>MIME type</a>) and a
+<dfn for="data: URL struct">body</dfn> (a <a>byte sequence</a>).
+
+<p>The <dfn export><code>data:</code> URL processor</dfn> takes a <a for=/>URL</a>
+<var>dataURL</var> and then runs these steps:
+
+<ol>
+ <li><p>Assert: <var>dataURL</var>'s <a for=url>scheme</a> is "<code>data</code>".
+
+ <li><p>Let <var>input</var> be the result of running the <a>URL serializer</a> on
+ <var>dataURL</var> with the <i>exclude fragment flag</i> set.
+
+ <li><p>Remove the leading "<code>data:</code>" string from <var>input</var>.
+
+ <li><p>Let <var>position</var> point at the start of <var>input</var>.
+
+ <li><p>Let <var>mimeType</var> be the result of <a>collecting a sequence of code points</a> that
+ are not equal to U+002C (,), given <var>position</var>.
+
+ <li><p>If <var>position</var> is past the end of <var>input</var>, then return failure.
+
+ <li><p>Advance <var>position</var> by 1.
+
+ <li><p>Let <var>encodedBody</var> be the remainder of <var>input</var>.
+
+ <li><p><a>Strip leading and trailing ASCII whitespace</a> from <var>mimeType</var>.
+
+ <li><p>Let <var>body</var> be the <a>string percent decoding</a> of <var>encodedBody</var>.
+
+ <li>
+  <p>If <var>mimeType</var> ends with an <a>ASCII case-insensitive</a> match for
+  "<code>;base64</code>", then:
+
+  <ol>
+   <li><p>Let <var>stringBody</var> be the <a>isomorphic decode</a> of <var>body</var>.
+
+   <li><p>Set <var>body</var> to the <a>forgiving-base64 decode</a> of <var>stringBody</var>.
+
+   <li><p>If <var>body</var> is failure, then return failure.
+
+   <li><p>Remove the last 7 code points from <var>mimeType</var>.
+  </ol>
+
+ <li><p>If <var>mimeType</var> starts with "<code>;</code>", then prepend "<code>text/plain</code>"
+ to <var>mimeType</var>.
+
+ <li><p>Let <var>mimeTypeRecord</var> be the result of <a lt="parse a MIME type">parsing</a>
```

Comment:
`<a lt="parse a MIME type">parsing</a>` is not a link in https://whatpr.org/fetch/579.html#data-url-struct-mime-type

Reply:
It'll go to https://mimesniff.spec.whatwg.org/#parse-a-mime-type tomorrow.